teuthology/orchestra/test/test_raise_for_status.py

```diff
@@ -0,0 +1,62 @@
+from io import BytesIO
+
+from mock import MagicMock, Mock
+from pytest import raises
+
+from teuthology.orchestra.run import RemoteProcess
+from teuthology.exceptions import (CommandCrashedError, CommandFailedError,
+                                   ConnectionLostError)
+
+
+class TestRaiseForStatus(object):
+    def setup(self):
+        m_transport = MagicMock()
+        m_transport.getpeername.return_value = ('name', 22)
+        self.m_ssh = MagicMock()
+        self.m_ssh.get_transport.return_value = m_transport
+        self.stdout = BytesIO(b'test_hostname')
+        self.stdout.seek(0)
+
+    def test_raise_for_status_failed(self):
+        proc = RemoteProcess(
+            client=self.m_ssh,
+            args=['foo'],
+            )
+        proc.stdout = self.stdout
+        proc._stdout_buf = Mock()
+        proc._stdout_buf.channel.recv_exit_status.return_value = 42
+        with raises(CommandFailedError) as exc:
```

Comment:
Not a bit fan of `with` statements but I can see that it is aligned with what is in https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/test/test_run.py
so this is good, always try to follow the convention of what came before unless it is super bad.